internal/controller/cloudflareaccessapplication_controller.go

```diff
@@ -127,9 +126,19 @@ func (r *CloudflareAccessApplicationReconciler) Reconcile(ctx context.Context, r
 		}
 	} else {
 		accessApp, err := api.AccessApplication(ctx, app.Status.AccessApplicationID)
-		existingaccessApp = &accessApp
 		if err != nil {
-			return ctrl.Result{}, errors.Wrap(err, "unable to get access application")
+			var apiErr *cloudflare.NotFoundError
+			if errors.As(err, &apiErr) {
+				log.Info("access application not found - recreating...", "accessApplicationID", app.Status.AccessApplicationID)
+				err := r.ClearStatus(ctx, app)
```

Comment:
❓ instead of clearing the status and persisting to the apiserver can we just set existingaccessApp to nil and clear out app.Status.AccessApplicationID?

that way it would avoid updating the status 2x in the same reconcile and we'll override it in ReconcileStatus